### PR TITLE
Fixes #163; Remove Helix.pitch and Helix.yaw

### LINKED ISSUE #163: remove Helix.pitch and Helix.yaw
Currently, both a HelixGroup and and Helix can have a pitch and yaw unequal to 0, but this is not recommended. It is recommended that only HelixGroups have nonzero pitch and yaw. If you want to make two Helices not parallel, the recommended way is to put them in separate HelixGroups, where all Helices in the HelixGroup are parallel.

This issue proposes a breaking change to codify this into a requirement by removing the fields Helix.yaw and Helix.pitch. Add methods to Design to look up the yaw and pitch of a Helix, which will delegate to the HelixGroup's property. But the only way make two helices non-parallel is to put them in separate groups.

The interpretation of the order in which to evaluate these (see #150 for explanation of this detail) remains yaw, then pitch, then roll. Although both a HelixGroup and Helix can have a nonzero roll, since it is applied last, it doesn't matter which order the two rolls are applied: the total roll of a Helix is simply `Helix.roll` + `HelixGroup.roll`.

### PR DESCRIPTION
**Pull request by UnHumbleBen**
# DO NOT MERGE YET

I still have not implemented a Design method for looking up pitch/yaw/roll values for helices. Should be straightforward, but any name suggestions for the function names? Maybe `Design.{pitch/yaw/roll}_of_helix`?

The second reason for not merging yet is because I added code in `Design.json` to allow for compatibility with designs that specified pitch and yaw in individual Helix. It passes all of the old test cases as well as news one I made, but I am concerned about the quality of the implementation so I would like to give @dave-doty an opportunity to review the code.

## Completed:
* Remove Helix.pitch and Helix.yaw
* Update Design.from_json to be compatible with designs that specified
  pitch and yaw in individual Helix (except when multiple helices in the
  same group have different pitch/yaw values, in which case,
  IllegalDesignError is raised)
* Updated examples that used pitch and roll
* Updated unit tests that used pitch/roll
* Added new unit test to test new Design.from_json

## Not Completed:
* Add Design method for looking up pitch/yaw/roll for helices

### CODE REVIEW

File: scadnano/scadnano.py
Comment (dave-doty) [inline scadnano/scadnano.py:3866]: I don't think this is a good way to test close floating point values. I've never seen this trick before (divide by the desired tolerance and then take a floor).

What if two values are `10.000001` and `9.999999` and `tolerance` is `0.01`? Then `10.000001 / 0.01` = `1000.0001`, which floors to `1000`, but `9.999999 / 0.01` = `999.9999`, which floors to `999`.

There would be similar problems with rounding to the nearest integer, e.g., if the two values were `1.004999` and `1.005001`, divided by `0.01` gives `100.4999` and `100.5001`, which would round to `100` and `101`, respectively, with `tolerance` = `0.01`, even though the numbers are within `0.01` of each other.

I think perhaps this logic could work, but you'd want to test whether adjacent integers are keys in the dict, e.g., if you convert a float to integer 1005, then you'd want any float converted to 1004 or 1006 to be considered equivalent. But I'm not sure if there's a concise way of expressing that.

I think the most readable approach, although it would be a bit more time-consuming (but this is small data, and the case where this is needed is rare), is to store the floating point values directly as keys, but then instead of asking `if yaw in pitch_yaw_to_helices`, instead do the linear-time loop over all keys `for existing_yaw in pitch_yaw_to_helices`, asking `if _is_close(yaw, existing_yaw)`.

File: tests/scadnano_tests.py
Comment (dave-doty) [inline tests/scadnano_tests.py:3547]: I didn't check the unit tests carefully, but make sure they are covering all the cases we outlined. (both Helix and Helix group have pitch specified, or neither, or one but not the other).

**Review by dave-doty — CHANGES_REQUESTED** [review]

**Comment by UnHumbleBen** [comment]
I made the final changes and ready to merge as soon as you approve. Let me know if you see anymore issues. One thing to note is that dev is technically in process of v0.15.2.

This PR would change dev version to v0.16.0. The only commit right now in dev is the one for removing the print statement. Should we just have that commit go under v0.16.0?

**Comment by dave-doty** [comment]
> I made the final changes and ready to merge as soon as you approve. Let me know if you see anymore issues. One thing to note is that dev is technically in process of v0.15.2.
> 
> This PR would change dev version to v0.16.0. The only commit right now in dev is the one for removing the print statement. Should we just have that commit go under v0.16.0?

Yes, just have that commit under 0.16. There's no need for a version 0.15.2.

File: scadnano/scadnano.py
Comment (dave-doty) [inline scadnano/scadnano.py:3794]: change comment to 

Same as the pitch of helix's :any:`HelixGroup`

File: scadnano/scadnano.py
Comment (dave-doty) [inline scadnano/scadnano.py:3964]: Put in some comments describing which portions of this code we expect to be only needed for backwards compatibility. It's a mix between code that will always be needed for import in v0.16.0 JSON, and code that will only be executed when encountering previous versions (e.g., v0.15.0) that stored pitch and yaw in Helix. Make sure it's easy to tell the difference between those two kinds of code by adding some comments saying when something is there for backward compatibility.

**Review by dave-doty — APPROVED** [review]

**Review by dave-doty — CHANGES_REQUESTED** [review]
Sorry, I accidentally pressed Approve instead of Request Changes last time. There's a couple comments I made requesting changes.

**Review by dave-doty — APPROVED** [review]

**Pull request merged**
